substrate/frame/beefy-mmr/src/lib.rs

```diff
@@ -258,17 +258,31 @@ where
 				},
 			};
 
-		let commitment_root =
-			match commitment.payload.get_decoded::<MerkleRootOf<T>>(&known_payloads::MMR_ROOT_ID) {
-				Some(commitment_root) => commitment_root,
+		let commitment_roots = commitment
+			.payload
+			.get_all_decoded::<MerkleRootOf<T>>(&known_payloads::MMR_ROOT_ID);
+		if commitment_roots.is_empty() {
+			// If the commitment doesn't contain any MMR root, while the proof is valid,
+			// the commitment is invalid
+			return true;
+		}
```

Comment:
If `get_all_decoded` returns a `Iterator` as I have proposed, this need to be replaced with a `bool found_valid = false`.